lib/ceedling/configurator.rb

```diff
@@ -193,17 +194,34 @@ def find_and_merge_plugins(config)
     end
 
     plugin_yml_defaults.each do |defaults|
-      @configurator_builder.populate_defaults( config, @yaml_wrapper.load(defaults) )
+      default_config.deep_merge!( @yaml_wrapper.load(defaults) )
     end
 
     plugin_hash_defaults.each do |defaults|
-      @configurator_builder.populate_defaults( config, defaults )
+      @configurator_builder.populate_defaults( default_config, defaults )
     end
 
     # special plugin setting for results printing
     config[:plugins][:display_raw_test_results] = true if (config[:plugins][:display_raw_test_results].nil?)
 
     paths_hash.each_pair { |name, path| config[:plugins][name] = path }
+    paths_hash.each_pair { |name, path| default_config[:plugins][name] = path } if (config[:plugins].nil?)
+  end
+
+
+  def populate_config_with_defaults(config, default_config)
+    @configurator_builder.populate_defaults( config, default_config )
+  end
+
+
+  # unity cmock initialilzer
+  def setup_unity_cmock_runner_config(config)
+    unity = config[:unity] || {}
+    @runner_config = unity.merge(@runner_config || config[:test_runner] || {})
+
+    cmock = config[:cmock] || {}
+    @runner_config = cmock.merge(@runner_config || config[:test_runner] || {})
```

Comment:
`@runner_config` is sure to be defined by this point.
```suggestion
    @runner_config = cmock.merge(@runner_config)
```